core/src/apps/cardano/byron_address.py

```diff
@@ -0,0 +1,116 @@
+from trezor import log
+from trezor.crypto import crc, hashlib
+
+from apps.cardano.protocol_magics import ProtocolMagics
+from apps.common import cbor
+from apps.common.seed import remove_ed25519_prefix
+
+if False:
+    from apps.cardano import seed
+
+
+"""
+This is the legacy implementation of Byron addresses. Byron
+addresses should however remain supported in Shelley with
+exactly the same implementation, thus it is kept here
+with base58 encoding and all the nuances of Byron addresses.
+"""
+
+
+def _encode_address_raw(address_data_encoded) -> bytes:
+    return cbor.encode(
+        [cbor.Tagged(24, address_data_encoded), crc.crc32(address_data_encoded)]
+    )
+
+
+def derive_byron_address(
+    keychain: seed.Keychain, path: list, protocol_magic: int
+) -> bytes:
+    node = keychain.derive(path)
+
+    # todo: GK - verify this is indeed correct (after IOHK confirmation/testnet support)
+    # protocol magic is included in Byron addresses only on testnets
+    if protocol_magic == ProtocolMagics.MAINNET:
+        address_attributes = {}
+        address_payload = None
+    else:
+        address_attributes = {2: cbor.encode(protocol_magic)}
+        address_payload = [address_attributes]
+
+    address_root = _get_address_root(node, address_payload)
+    address_type = 0
```

Comment:
you can use the CardanoAddressType.BYRON_ADDRESS constant here

Reply:
I could, but I wouldn't want to mix those two things as they are unrelated and only coincidentally the same. 

It doesn't matter anymore, since I've merged address_id and address_type to be the same thing and Byron address now has the value of 8.